### PR TITLE
Add runner params CLI option

### LINKED ISSUE #809: Limit the number of parallel nodes
## Description
I would like to be able to limit the number of nodes running in parallel when I use `kedro run --parallel`

## Context
This feature is important to avoid memory error problems due to memory size limitations. 
For instance, I have a pipeline composed of many independent nodes with big dataset inputs (more than 1 GB each). So, when I run kedro specifying the `--parallel` option, all these node starts simultaneously and load all inputs, reaching the memory limit. 

If we could specify the number of concurrent nodes, we will be able to control de memory usage.

### PR DESCRIPTION
**Pull request by BALOGUN-DAVID**
## Description

Closes #809.

This PR adds a `--runner-params` option to `kedro run`, allowing runner constructor keyword arguments to be passed from the CLI.

This makes it possible to configure runner-specific options such as `max_workers` without overriding the project CLI:

```bash
kedro run --runner=ParallelRunner --runner-params=max_workers=4
```

## Development notes

Implemented `--runner-params` for `kedro run` and wired the parsed values into runner construction.

Changes include:

- Added `--runner-params` to the `kedro run` command.
- Reused the existing CLI params parser for values such as `max_workers=4,is_async=True`.
- Added runner keyword resolution so the existing `--async` flag remains supported for compatibility.
- Deprecated `--async` in favour of `--runner-params=is_async=True`.
- Added a `KedroCliError` when `--async` conflicts with `--runner-params=is_async=False`.
- Updated runner documentation, the custom CLI example, and `RELEASE.md`.
- Added tests for CLI runner params, config-file runner params, default async behaviour, and the conflict case.

Tested with:

```bash
.venv\Scripts\python.exe -m pytest --no-cov tests\framework\cli\test_cli.py -q
.venv\Scripts\pre-commit.exe run ruff --files kedro\framework\cli\project.py tests\framework\cli\test_cli.py docs\build\run_a_pipeline.md docs\getting-started\commands_reference.md RELEASE.md
.venv\Scripts\pre-commit.exe run ruff-format --files kedro\framework\cli\project.py tests\framework\cli\test_cli.py
.venv\Scripts\pre-commit.exe run trailing-whitespace --files kedro\framework\cli\project.py tests\framework\cli\test_cli.py docs\build\run_a_pipeline.md docs\getting-started\commands_reference.md RELEASE.md
.venv\Scripts\pre-commit.exe run end-of-file-fixer --files kedro\framework\cli\project.py tests\framework\cli\test_cli.py docs\build\run_a_pipeline.md docs\getting-started\commands_reference.md RELEASE.md
.venv\Scripts\python.exe -m mypy kedro\framework\cli\project.py --strict --allow-any-generics --no-warn-unused-ignores
git diff --check
```

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/signing-commits)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [x] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Upcoming Release
 ## Major features and improvements
+* Added `--runner-params` to `kedro run`, allowing runner constructor keyword arguments such as `max_workers` to be passed from the CLI.
```
Comment (ElenaKhaustova) [inline RELEASE.md:3]: **Suggestion:** Consider adding a second bullet under "Bug fixes and other changes" noting the `--async` deprecation, e.g.:

```markdown
* Deprecated `--async` flag for `kedro run` in favour of `--runner-params=is_async=True`.
```

This gives users scanning the changelog a heads-up about the deprecation alongside the new feature.

**Review by ElenaKhaustova — COMMENTED** [review]
## Kedro PR Review
> Generated with `review-kedro-pr` skill.

### Overview
- **Critical:** 0 | **Suggestions:** 3

Great work on this PR — it aligns well with the [team's agreed approach](https://github.com/kedro-org/kedro/issues/809#issuecomment-4545458411) and the implementation is sound. The `_resolve_runner_kwargs` logic correctly handles all combinations of `--async` and `--runner-params`. Backward compatibility is preserved and the reuse of `_split_params` for parsing is a good design choice.

See inline suggestions below.

### Suggestions (consider improving)
- [ ] `RELEASE.md:L3` — Consider adding a bullet noting the `--async` deprecation
- [ ] `kedro/framework/cli/project.py:L330` — Consider specifying a target version for the deprecation removal
- [ ] `tests/framework/cli/test_cli.py` — Consider adding an edge-case test for invalid runner params

### Notes
- **General review:** Clean, focused PR. Logic is correct. No regressions.
- **PR checklist:** RELEASE.md entry present. Docs updated. Tests added. Community Contributions credit to be added by maintainer.
- **API usage:** Runner construction follows the existing pattern correctly.

File: kedro/framework/cli/project.py [hunk 2]
```diff
@@ -305,3 +317,27 @@ def run(  # noqa: PLR0913
...
+
+def _resolve_runner_kwargs(
+    is_async: bool, runner_params: dict[str, Any]
+) -> dict[str, Any]:
+    """Merge runner params with the legacy --async flag."""
+    runner_kwargs = dict(runner_params)
+
+    if is_async:
+        warnings.warn(
+            "Option '--async' is deprecated and will be removed in a future release. "
```
Comment (ElenaKhaustova) [inline kedro/framework/cli/project.py]: **Suggestion:** This follows the existing pattern (the `--pipeline` deprecation uses the same wording), but ideally deprecation warnings should state *when* the removal will happen so users can plan. Consider:

```python
"Option '--async' is deprecated and will be removed in Kedro 2.0. "
```

(Or whatever the target version is — the team's issue comment says "the next breaking release".)

**Comment by BALOGUN-DAVID** [comment]
Thanks for the review! I’ve addressed the suggestions:

- Added a `RELEASE.md` bullet for the `--async` deprecation.
- Updated the deprecation warning to state removal in Kedro 2.0.
- Updated the async docs wording to match.
- Added an invalid `--runner-params` CLI test case.

Verified with the targeted CLI tests, ruff, ruff-format, whitespace/end-of-file hooks, mypy on `project.py`, and `git diff --check`.

**Review by ElenaKhaustova — APPROVED** [review]
LGTM, thank you @BALOGUN-DAVID!

File: RELEASE.md [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 * Added the node function name to project inspection snapshots as `NodeSnapshot.func_name`.
 
 ## Bug fixes and other changes
 * Fixed a `RecursionError` when initialising a session with dynaconf-backed settings by converting `settings.SESSION_STORE_ARGS` to a plain `dict` before deepcopying it.
 * Excluded `kedro_benchmarks` from the built wheel so benchmark tests are no longer shipped with the package.
 * Fixed `get_close_matches` returning duplicate suggestions when several inputs matched the same target, and being able to exhaust a one-shot iterable passed as `targets`.
+* Deprecated `--async` flag for `kedro run` in favour of `--runner-params=is_async=True`.
 
 ## Documentation changes
 ## Community contributions
```
Comment (ravi-kumar-pilla) [inline RELEASE.md:15]: Thanks for the contribution @BALOGUN-DAVID . Can you please add yourself to the Community contributions section here.

**Review by ravi-kumar-pilla — APPROVED** [review]
